knative-operator/pkg/controller/knativekafka/knativekafka_controller_test.go

```diff
@@ -0,0 +1,500 @@
+package knativekafka
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	mf "github.com/manifestival/manifestival"
+	v1alpha1 "github.com/openshift-knative/serverless-operator/knative-operator/pkg/apis/operator/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	admissiontypes "sigs.k8s.io/controller-runtime/pkg/webhook/admission/types"
+)
+
+var (
+	defaultRequest = reconcile.Request{
+		NamespacedName: types.NamespacedName{Namespace: "knative-eventing", Name: "knative-kafka"},
+	}
+)
+
+type fakeManager struct {
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+func (*fakeManager) Add(manager.Runnable) error {
+	return nil
+}
+
+func (*fakeManager) SetFields(interface{}) error {
+	return nil
+}
+
+func (*fakeManager) Start(<-chan struct{}) error {
+	return nil
+}
+
+func (*fakeManager) GetConfig() *rest.Config {
+	return nil
+}
+
+func (f *fakeManager) GetScheme() *runtime.Scheme {
+	return f.scheme
+}
+
+func (*fakeManager) GetAdmissionDecoder() admissiontypes.Decoder {
+	return nil
+}
+
+func (f *fakeManager) GetClient() client.Client {
+	return f.client
+}
+
+func (*fakeManager) GetFieldIndexer() client.FieldIndexer {
+	return nil
+}
+
+func (*fakeManager) GetCache() cache.Cache {
+	return nil
+}
+
+func (*fakeManager) GetRecorder(name string) record.EventRecorder {
+	return nil
+}
+
+func (*fakeManager) GetRESTMapper() meta.RESTMapper {
+	return nil
+}
```

Comment:
I just pushed `fakeManager` within this file. If you do not know an existing fake manager like this one, I will extract this into a separate file. If you do know, please let me know and I will get rid of this.

Reply:
I would refactor the `newReconciler` function to take a scheme and a client rather than the manager. That obviates this completely, I think.

Reply:
Yeah, that's an option, I actually tried it.
But I ended up loading the manifest more than once from the file, so I gave up.

I am open to suggestions for that.

Reply:
@markusthoemmes 
Got rid of `fakeManager` now by loading the manifests within the unit test.

Mind having another look?